### ISSUE TITLE
images/caches created in src/...

### ISSUE THREAD

**Issue by ascheman** [asciidoctor/asciidoctor-maven-plugin#186]
The AsciiDoctor Maven Plugin creates files (images and caches) in the source location of .adoc files, somewhere beyond src/. Any file generated by Maven should be created in target/ only.

**Comment by mojavelinux** [comment]
This is not something the Maven Plugin controls but rather something that whatever is generating the files is responsible for. If you are referring to Asciidoctor Diagram, then the latest version now puts the generated files into the target directory by default. See https://github.com/asciidoctor/asciidoctor-maven-examples/tree/master/asciidoctor-diagram-example

**Comment by ascheman** [comment]
Right, sorry, switching to the latest asciidoctor-diagram ruby gem (1.5.3) solved the problem!

**Comment by mojavelinux** [comment]
Excellent!

**Comment by ascheman** [comment]
If I add the ```:data-uri:``` pragma to the mentioned sample or to my own adocs, I do see the following error

```
...
asciidoctor: WARNING: image to embed not found or not readable: /Users/ascheman/wrk/dukecon/asciidoctor-maven-examples/asciidoctor-diagram-example/src/docs/asciidoc/images/asciidoctor-diagram-process.png
asciidoctor: WARNING: image to embed not found or not readable: /Users/ascheman/wrk/dukecon/asciidoctor-maven-examples/asciidoctor-diagram-example/src/docs/asciidoc/images/auth-protocol.png
asciidoctor: WARNING: image to embed not found or not readable: /Users/ascheman/wrk/dukecon/asciidoctor-maven-examples/asciidoctor-diagram-example/src/docs/asciidoc/images/dot-example.svg
[INFO] Rendered /Users/ascheman/wrk/dukecon/asciidoctor-maven-examples/asciidoctor-diagram-example/src/docs/asciidoc/example-manual.adoc
...
```

This didn't happen before switching to the latest versions ... Is this a new bug?

**Comment by mojavelinux** [comment]
@ascheman There's a catch-22 here. The root problem is that we simply don't have enough attributes to describe the different image locations for all the scenarios (i.e., they are not expressive enough).

The problem is that Asciidoctor Diagram is generating images into a folder with the intent of linking to them. When Asciidoctor tries to embed the image, it doesn't know anything about this `imagesoutdir` location and instead tries to resolve them from `imagesdir`. Thus, you have to set the `imagesdir` to tell Asciidoctor where to find the generated images. The drawback, however, is that then it won't be able to find the other images to embed.

A workaround for this is to toggle the `imagesdir` attribute value, described in the following issue. https://github.com/asciidoctor/asciidoctor-pdf/issues/271#issuecomment-166569803

To arrive at a real solution, we really need to go back to the drawing board regarding image paths and try to consider the scenarios we now have. Because we just don't have enough coverage with the existing design to handle all the cases.
